client/modules/User/components/ExamplePreview.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { PropTypes } from 'prop-types';
+import axios from 'axios';
+import { createGlobalStyle } from 'styled-components';
+import getConfig from '../../../utils/getConfig';
+import EmbedFrame from '../../Preview/EmbedFrame';
+import { registerFrame } from '../../../utils/dispatcher';
+
+const GlobalStyle = createGlobalStyle`
+  body {
+    margin: 0;
+  }
+`;
+
+function ExamplePreview({ url }) {
+  const [sketch, setSketch] = useState(null);
+  registerFrame(window.parent, getConfig('EDITOR_URL'));
+
+  useEffect(() => {
+    async function getRandomProject() {
+      // Find example projects
+      const response = await axios.get(url);
+      const projects = response.data;
+      if (!projects) return null;
+      setSketch(projects);
+      console.log('sketch', sketch);
+      return projects;
+    }
+    getRandomProject();
+  }, [url]);
+
+  const files = useMemo(() => {
+    const instanceMode = (sketch?.files || [])
+      .find((file) => file.name === 'sketch.js')
+      ?.content?.includes('Instance Mode');
+
+    return (sketch?.files || []).map((file) => ({
+      ...file,
+      content: file.content
+        // Fix links to assets
+        .replace(
+          /'assets/g,
+          "'https://github.com/processing/p5.js-website/raw/main/src/data/examples/assets"
+        )
+        .replace(
+          /"assets/g,
+          '"https://github.com/processing/p5.js-website/raw/main/src/data/examples/assets'
+        )
```

Comment:
hmmm...this will only work for sketches on the p5 user.  But the code in `SketchList` adds this functionality to all users' sketches.  So we would need to get the files from s3.  We have code to do that in other places but it's really complex.  I think it's already handled inside the `EmbedFrame` though?